src/Microsoft.Health.ControlPlane.Core/Features/Rbac/IdentityProvider.cs

```diff
@@ -0,0 +1,31 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.Health.ControlPlane.Core.Features.Rbac
+{
+    public class IdentityProvider
+    {
+        public IdentityProvider()
```

Comment:
We should make the class immutable unless there is a need to have default constructor and have all of the properties settable from outside. If this is for the serialization, there is a way to work around this without making the class mutable.

Reply:
Made immutable.